Scenario load tests with simulator

itstauq opened the pull request:
## Description

Closes https://github.com/OpenMined/Heartbeat/issues/1863, https://github.com/OpenMined/Heartbeat/issues/1862

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

itstauq committed:
Add scenario tests to the CI

itstauq committed:
add support to run scenario tests using mock apis

itstauq committed:
Clean up 1

itstauq committed:
Refactor

khoaguin committed:
Merge branch 'dev' into tauquir/scenario-tests

yashgorana committed:
some early l0 activities in sim

yashgorana committed:
some more code

abyesilyurt committed:
exit early if an exception occurs

abyesilyurt committed:
revert sleep duration

khoaguin committed:
working on low-side ds workflow

yashgorana committed:
Merge branch 'dev' into tauquir/scenario-tests

khoaguin committed:
user on low-side submits queries and waiting for approval

khoaguin committed:
[tox] create tox task for k8s bigquery sync scenario test

khoaguin committed:
fix wrong import

yashgorana committed:
register users

yashgorana committed:
Merge branch 'dev' into tauquir/scenario-tests

yashgorana committed:
use unsync_guard

yashgorana committed:
polish things up, separately

yashgorana committed:
l0 till bigquery.test_query

yashgorana committed:
logs

yashgorana committed:
to_thread worker pool create

yashgorana committed:
l0 submit_query & sync

khoaguin committed:
Merge branch 'dev' into tauquir/scenario-tests

khoaguin committed:
fix lint

khoaguin committed:
[tests/scenario] level 2 bigquery scenario test version 2

yashgorana committed:
Merge branch 'dev' into tauquir/scenario-tests

yashgorana committed:
l0 admin high triage requests

yashgorana committed:
Merge branch 'dev' into tauquir/scenario-tests

yashgorana committed:
Merge branch 'dev' into tauquir/scenario-tests

khoaguin committed:
[[tests/scenario] admin low triggers signal when all results are available so admin high can stop triaging

khoaguin committed:
[test/scenarios] admin low breaking from watching sync coroutine after receiving all results (synced from high)

khoaguin committed:
[tests/scenarios] add a coroutine for users to check results after they are available

khoaguin committed:
[tests/scenarios] logs are now saved at /scenariosv2/logs/{date_time}

khoaguin committed:
Merge branch 'dev' into tauquir/scenario-tests

yashgorana committed:
fix 'worker_pool_name' var name change

yashgorana committed:
drop F841

khoaguin committed:
[CI] installing `just` and running bigquery scenariov2 on k8s with just commands

khoaguin committed:
[CI] fix bash commands for `syft.test.scenario`

khoaguin committed:
[CI] move testing simulation scenario tests to pr-tests-stack
[tox] clean up & fix stack.test.scenario.k8s task

khoaguin committed:
[CI] checking python version

yashgorana committed:
Merge branch 'dev' into tauquir/scenario-tests

yashgorana committed:
drop scenarios

yashgorana committed:
tweaked logging

yashgorana committed:
attempt to make things common

yashgorana committed:
rename bq api user flows

yashgorana committed:
refactor bq_create_pool

yashgorana committed:
register users concurrently

yashgorana committed:
admin api concurrency

yashgorana committed:
de-nest sim_entrypoint wrapper

yashgorana committed:
admin sync refactor

yashgorana committed:
add .logs ignore

yashgorana commented on packages/grid/helm/examples/gcp/gcp.high.yaml:
@khoaguin  why did we change this? I updated this a week or two ago to make sure gcp.high is without a bucket mount.

yashgorana replied:
oh it looks like some merge conflict came through

khoaguin replied:
yep I didn't touch this part of the code

yashgorana committed:
fix merge conflicts

yashgorana committed:
drop unsync

khoaguin committed:
remove unused import. upper case TODO

khoaguin committed:
Merge branch 'dev' into tauquir/scenario-tests

khoaguin committed:
[tests/scenarios] always print side-type after "Admin" in logs

khoaguin committed:
[tests/scenarios] adding a try catch in `admin_sync`

khoaguin committed:
[tox] trying to fix `No interpreter found for Python 3.12 in system toolchains` issue on CI

khoaguin committed:
[CI] install just to `/usr/local/bin`

khoaguin committed:
[tox] change `deps` for `stack.test.scenario.k8s`

khoaguin committed:
[CI] update uv, tox and uv-tox version

khoaguin committed:
[CI] update uv and tox to latest versions for all tasks

yashgorana commented on tox.ini:
```diff
@@ -291,21 +292,45 @@ commands =
 
-[testenv:syft.test.scenario]
-description = Syft Scenario Tests
+[testenv:stack.test.scenario.k8s]
```
@khoaguin a couple of things for this commands

- Let's separate L0 & L2 test runs... at-least that way whenever a PR test fails, we know what is failing. Open to what you feel about this
- I was thinking of starting to make our tox tidier by leveraging the new features. We can move cluster setup code to `commands_pre` and instead of doing a bash -c, we can actually stack all the commands in a single run i.e.
```
commands_pre = 
    just delete-all start-high deploy-high
```
- We can add the teardown in `commands_post` by simply running `just delete-all`
- Now that we've added startupProbes/livenessProbes to pods, we can actually wait for the pods by simply calling the following instead of multiple `wait_for.sh`
```
kubectl wait --for=condition=ready pod --all --timeout=300s --namespace <ns> --context <ctx>
```
Infact we can add a new just command under utils group called
```
[group(...
wait-pods kube_context namespace:
  @kubectl...
```
that runs the above command or the multiple wait_for. Then we extend it for each cluster so that we have `wait-high: (wait-pods _ctx_high _ns_default)`, `wait-low` etc

- Finally, we can teardown everything in `commands_post` by calling `just delete-all`

yashgorana replied:
Also how do we do python-server testing?

khoaguin replied:
Thanks for the feedback @yashgorana. Now we have 2 separate tox and CI tasks for L0 and L2 (k8s). All using the new `just` commands, with `commands_pre` and `commands_post`
I made [this ticket](https://github.com/OpenMined/Heartbeat/issues/1874) to clean the whole `tox.ini`

khoaguin replied:
Working on python-server testing.

yashgorana commented on .github/workflows/pr-tests-stack.yml:
```diff
@@ -53,7 +53,7 @@ jobs:
         run: |
           python -m pip install --upgrade pip
-          pip install uv==0.4.1 tox==4.18.0 tox-uv==1.11.2
+          pip install uv==0.4.16 tox==4.20.0 tox-uv==1.13.0
```
let's not bump any version for now :) Minimize any variables that may cause problems

khoaguin replied:
thanks Yash. Reverted it back

yashgorana committed:
just wait-pods

khoaguin committed:
[CI] reverted to use `uv==0.4.1 tox==4.18.0 tox-uv==1.11.2`

khoaguin committed:
[tox] separate simulation k8s test for L0 and L2
- use new `just` commands for everything
[CI] separate simulation k8s test for L0 and L2

khoaguin committed:
[CI] clean up k3d for simulation k8 tests

khoaguin committed:
[test/scenarios] launching python server if testing on python

khoaguin committed:
[tests/scenarios] launching python nodes with producer and consumers
[just] trying to fix `wait-pods` error: `no matching resources found`

yashgorana commented on .github/workflows/pr-tests-stack.yml:
```diff
@@ -956,3 +956,256 @@ jobs:
+          devspace version
+          tox -e stack.test.scenario.k8s
+
+      # todo: collect logs
```
yeah this is important

khoaguin replied:
it's done @yashgorana

khoaguin committed:
[tests/scenario] l0 and l2 bigquery tests work with python servers

khoaguin committed:
[CI] fix the ORCHESTRA_DEPLOYMENT_TYPE env variable set to wrong value

khoaguin committed:
Merge branch 'dev' into tauquir/scenario-tests

khoaguin committed:
[CI] fix tox command for scenario k8s test

khoaguin committed:
[CI] re-enable `test-syft-scenario` and add `test-syft-scenario-sync` task

khoaguin committed:
[tox] add necessary pip packages for syft scenario tests

yashgorana committed:
DeploymentType.REMOTE as default - k8s first

yashgorana committed:
refactor orchestra stuff

khoaguin committed:
[tests/scenario] fix small bug
[tox] add `ORCHESTRA_DEPLOYMENT_TYPE` for scenario python tests

khoaguin committed:
[CI] collecting logs for `tests-simulation-scenario-k8s`

khoaguin committed:
[CI] collecting logs for `tests-simulation-scenario-sync-k8s`

khoaguin committed:
[CI] collecting logs for syft-scenario and syft-scenario-sync tests

abyesilyurt committed:
fix ActionObject.get_sync_dependencies

khoaguin committed:
Merge branch 'dev' into tauquir/scenario-tests

khoaguin committed:
[tests/scenarios] increase timeout for l0

abyesilyurt committed:
test with 5 users

abyesilyurt committed:
Merge remote-tracking branch 'origin/fix_status_l0' into tauquir/scenario-tests

abyesilyurt committed:
update code call

abyesilyurt committed:
increase timeout

abyesilyurt committed:
increase test timeout

khoaguin committed:
Merge branch 'dev' into tauquir/scenario-tests

abyesilyurt committed:
Merge remote-tracking branch 'origin' into tauquir/scenario-tests

abyesilyurt committed:
Merge remote-tracking branch 'origin' into tauquir/scenario-tests

Commit:
Merge branch 'dev' into tauquir/scenario-tests

abyesilyurt committed:
Merge branch 'tauquir/scenario-tests' of github.com:OpenMined/PySyft into tauquir/scenario-tests

abyesilyurt committed:
fix

abyesilyurt committed:
fix

abyesilyurt committed:
fix results

Commit:
Merge branch 'dev' into tauquir/scenario-tests

abyesilyurt committed:
request.code.__call__ uses the same api as code.funcname

abyesilyurt committed:
Merge branch 'tauquir/scenario-tests' of github.com:OpenMined/PySyft into tauquir/scenario-tests

khoaguin committed:
Merge branch 'dev' into tauquir/scenario-tests
